src/main/java/org/reaktivity/nukleus/maven/plugin/internal/generate/UnionFlyweightGenerator.java

```diff
@@ -897,20 +897,22 @@ private void addClassType(
                 if ("StringFW".equals(className.simpleName()) || "String16FW".equals(className.simpleName()))
                 {
                     CodeBlock.Builder codeBlock = CodeBlock.builder();
+                    ClassName builderType = className.nestedClass("Builder");
 
                     // TODO: handle optional fields
                     codeBlock.beginControlFlow("if (value == null)")
                         .addStatement("limit(offset() + $L)", offset(name))
                         .nextControlFlow("else")
                         .addStatement("kind($L)", kind(name))
-                        .addStatement("$L().set(value, $T.UTF_8)", name, StandardCharsets.class);
+                        .addStatement("$T $L = $L()", builderType, name, name)
+                        .addStatement("$L.set(value, $T.UTF_8)", name, StandardCharsets.class);
 
                     if (nextType instanceof ParameterizedTypeName)
                     {
-                        codeBlock.addStatement("$L($L().build().limit())", nextName, name);
+                        codeBlock.addStatement("$L($L.build().limit())", nextName, name);
```

Comment:
You asked how to test this. I think we would get here if the following field is a parameterized type, like List<x>. Try adding a test union with that characteristic (a list<x> field after the string) and see if it takes you here.

Reply:
Agreed, the `list` type is parametric so should trigger this usecase for a union with a `list` field after a `string` field.

Reply:
@jfallows, @cmebarrow Thanks for the feedback! I have tried to add the test case but have hit following issue: https://github.com/reaktivity/nukleus-maven-plugin/issues/40. Maybe this PR can be merged like this and enhance the test when fixing that issue ?